docs/source/plugin/time.rst

```diff
@@ -0,0 +1,211 @@
+    datetimes. You can always assume it is available for your plugin since
+    Sopel depends on this library.
+
+.. __: https://pypi.org/project/pytz/
+
+Format time
+===========
+
+So far, you have:
+
+* an aware datetime in UTC+0
+* the user (or channel) timezone
+
+And you want to:
+
+* display the time properly formatted for a user/channel
+
+Then you have arrived at the last step of your journey, thanks to the
+:func:`~sopel.tools.time.format_time` function::
+
+    >>> from sopel.tools.time import format_time
+    >>> format_time(
+    ...  bot.db, bot.settings,
+    ...  zone=custom_tz, nick='Nick', channel='#sopel', time=now,
+    ... )
+    '2021-07-26 - 18:07:49 (Europe/Paris)'
+
+And voilà! You now have a string formatted aware datetime that uses the format
+defined for a user/channel/the bot, and can now rest and enjoy your own time.
+
+Best practices
+==============
+
+So far, you have learn how to get a time formatted with the preferred timezone
+and format of a user: this is perfect for a command like ``.time`` that
+displays the time for a specific user in mind. However, other commands, like
+URL previews, are not related to users, and they should use a different
+strategy to figure out the right timezone and format.
+
+User specific time
+------------------
+
+A user specific time is when the time is displayed for a specific user: a
+direct message, a reminder, the user's time, etc.
+
+In that case, the recommended order to select the appropriate timezone and time
+format is:
+
+* user's preferred ones
+* channel's preferred ones
+* bot's timezone and format (from configuration)
+* default timezone and format
+
+This can be done with :func:`~sopel.tools.time.get_timezone` and
+:func:`~sopel.tools.time.format_time`::
+
+    >>> custom_tz = get_timezone(
+    ...     bot.db, bot.settings,
+    ...     zone=None, nick=nick_name, channel=channel_name,
+    ... )
+    >>> display_time = format_time(
+    ...     bot.db, bot.settings, zone=custom_tz,
+    ...     nick=nick_name, channel=channel_name, time=user_time
+    ... )
+
+Other times
+-----------
+
+When displaying a time that is not for a specific user, it doesn't make sense
+to display time with the user's preferred format. For example, a URL preview
+plugin is not displaying a user specific time.
+
+In that case, the recommended order to select the appropriate timezone and time
+format is:
+
+* channel's preferred ones
+* bot's timezone and format (from configuration)
+* default timezone and format
+
+This can be done with :func:`~sopel.tools.time.get_timezone` and
+:func:`~sopel.tools.time.format_time`::
+
+    >>> custom_tz = get_timezone(
+    ...     bot.db, bot.settings,
+    ...     zone=None, channel=channel_name,
+    ... )
+    >>> display_time = plugin_defined_format or format_time(
+    ...     bot.db, bot.settings, zone=custom_tz,
+    ...     channel=channel_name, time=plugin_time
+    ... )
```

Comment:
```suggestion
    >>> custom_tz = get_timezone(
    ...     bot.db, bot.settings, zone=None,
    ...     channel=channel_name,
    ... )
    >>> display_time = plugin_defined_format or format_time(
    ...     bot.db, bot.settings, zone=custom_tz,
    ...     channel=channel_name, time=plugin_time,
    ... )
```

Here, and above, I get you're trying to balance the line lengths without splitting each argument onto its own line—but I say it's more important to make the omitted `nick` param obvious, and it's most obviously missing (here) if it's in the same place on the line (above) in both function calls.